internal/dag/httpproxy_processor.go

```diff
@@ -331,6 +343,78 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 					}
 				}
 			}
+
+			providerNames := sets.NewString()
+			for _, jwtProvider := range proxy.Spec.VirtualHost.JWTProviders {
+				if providerNames.Has(jwtProvider.Name) {
+					validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "DuplicateProviderName",
+						"Spec.VirtualHost.JWTProviders is invalid: duplicate name %s", jwtProvider.Name)
+					return
+				}
+				providerNames.Insert(jwtProvider.Name)
+
+				if jwtProvider.Default {
+					if len(defaultJWTProvider) > 0 {
+						validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "MultipleDefaultProvidersSpecified",
+							"Spec.VirtualHost.JWTProviders is invalid: at most one provider can be set as the default")
+						return
+					}
+					defaultJWTProvider = jwtProvider.Name
+				}
+
+				jwksURL, err := url.Parse(jwtProvider.RemoteJWKS.URI)
+				if err != nil {
+					validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "RemoteJWKSURIInvalid",
+						"Spec.VirtualHost.JWTProviders.RemoteJWKS.URI is invalid: %s", err)
+					return
+				}
+
+				if jwksURL.Scheme != "http" && jwksURL.Scheme != "https" {
+					validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "RemoteJWKSSchemeInvalid",
+						"Spec.VirtualHost.JWTProviders.RemoteJWKS.URI has invalid scheme %q, must be http or https", jwksURL.Scheme)
+					return
+				}
+
+				jwksTimeout := time.Second
+				if len(jwtProvider.RemoteJWKS.Timeout) > 0 {
+					res, err := time.ParseDuration(jwtProvider.RemoteJWKS.Timeout)
+					if err != nil {
+						validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "RemoteJWKSTimeoutInvalid",
+							"Spec.VirtualHost.JWTProviders.RemoteJWKS.Timeout is invalid: %s", err)
+						return
+					}
+
+					jwksTimeout = res
+				}
+
+				var cacheDuration *time.Duration
+				if len(jwtProvider.RemoteJWKS.CacheDuration) > 0 {
+					res, err := time.ParseDuration(jwtProvider.RemoteJWKS.CacheDuration)
+					if err != nil {
+						validCond.AddErrorf(contour_api_v1.ConditionTypeJWTVerificationError, "RemoteJWKSCacheDurationInvalid",
+							"Spec.VirtualHost.JWTProviders.RemoteJWKS.CacheDuration is invalid: %s", err)
+						return
+					}
+
+					cacheDuration = &res
+				}
+
+				svhost.JWTProviders = append(svhost.JWTProviders, JWTProvider{
+					Name:      jwtProvider.Name,
+					Issuer:    jwtProvider.Issuer,
+					Audiences: jwtProvider.Audiences,
+					RemoteJWKS: RemoteJWKS{
+						URI:     jwtProvider.RemoteJWKS.URI,
+						Timeout: jwksTimeout,
+						Cluster: DNSNameCluster{
+							Address:         jwksURL.Hostname(),
+							Scheme:          jwksURL.Scheme,
+							DNSLookupFamily: string(p.DNSLookupFamily),
+						},
```

Comment:
Shouldn't we allow user to set non-default port number for jwksURL? That is, should we add Port field that can override default 80 and 443?

Reply:
Yep, we should definitely support that.

Reply:
Implemented in https://github.com/projectcontour/contour/pull/4723/commits/479c8099f0dd76d756ac31c0d44b3d8dd8d0a4e1